arbor/partition_load_balance.cpp

```diff
@@ -90,11 +125,11 @@ domain_decomposition partition_load_balance(
                     q.pop();
                     cg.push_back(element);
                     // Adjacency list
-                    auto conns = rec.gap_junctions_on(element);
-                    for (auto c: conns) {
-                        if (!visited.count(c.peer.gid)) {
-                            visited.insert(c.peer.gid);
-                            q.push(c.peer.gid);
+                    auto conns = global_gj_connection_table[element];
+                    for (const auto& peer: conns) {
```

Comment:
To avoid making a copy of `global_gj_connection_table[element]`, which is a `std::vector`, and looking up `peer` twice in the `visited` set, consider the following:

```C++
for (const auto& peer: global_gj_connection_table[element]) {
  if (visited.insert(peer).second) {
    q.push(peer);
  }
}
```

Reply:
I think that was supposed to be `const auto& conns = global_gj_connection_table[element]`. I will implement your suggestion anyway.

Reply:
I actually meant what I proposed: there is no need to create the `const auto& cons`, though whether you do or don't is a matter of personal preference.